plugins/calibre.koplugin/wireless.lua

```diff
@@ -644,9 +644,20 @@ end
 
 function CalibreWireless:sendToCalibre(arg)
     logger.dbg("GET_BOOK_FILE_SEGMENT", arg)
-    -- not implemented yet, we just send an invalid opcode to raise a control error in calibre.
-    -- If we don't do this calibre will wait *a lot* for the file(s)
-    self:sendJsonData('NOOP', {})
+    local inbox_dir = G_reader_settings:readSetting("inbox_dir")
+    local path = inbox_dir .. "/" .. arg.lpath
+
+    local file = io.open(path, "rb")
+    if file == nil then
+        self:sendJsonData('NOOP', {})
+        return
+    end
+
+    local data = file:read("*all")
```

Comment:
Is it possible to split the binary in chunks of a few KB/MB?. That way the available memory isn't a hard limit. We might want to prevent OOM errors as 1GB files are not that rare.

When sending files from calibre to koreader we already do that. See https://github.com/koreader/koreader/blob/master/plugins/calibre.koplugin/wireless.lua#L540-L579.

I guess it works the same way here.

ie: use a buffer of 4096. Read the whole file to know its size and store it in a `to_write_bytes` local variable. Then send the `ok, file size` .  Then use file:seek to move between offsets at the file and send chunks until `to_write_bytes` is equal to 0.

I'm not sure how calibre deals with it, but I'm guessing it receives the bytes count first and reads from the socket until that number of bytes is reached.

Reply:
Correct, calibre receives the byte count and then tries to read that number of bytes from the socket.

You're right, we should chunk the data. As usual, I was only thinking of the scope of my own problem.
I was trying to deal with the "sync stats" plugin, which only transfers the `metadata.[filetype].lua` files back to calibre. I hope that 1GB `.lua` files are rare ;)

Reply:
Simple chunking added